src/Tracking/Conversions.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Pinterest for WooCommerce Tracking. Conversions API.
+ *
+ * @package     Pinterest_For_WooCommerce/Classes/
+ * @version     1.0.0
+ */
+
+namespace Automattic\WooCommerce\Pinterest\Tracking;
+
+use Automattic\WooCommerce\Pinterest\API\APIV5;
+use Automattic\WooCommerce\Pinterest\Logger as Logger;
+use Automattic\WooCommerce\Pinterest\Tracking;
+use Automattic\WooCommerce\Pinterest\Tracking\Data\Category;
+use Automattic\WooCommerce\Pinterest\Tracking\Data\Checkout;
+use Automattic\WooCommerce\Pinterest\Tracking\Data\None;
+use Automattic\WooCommerce\Pinterest\Tracking\Data\Product;
+use Automattic\WooCommerce\Pinterest\Tracking\Data\Search;
+use Automattic\WooCommerce\Pinterest\Tracking\Data\User;
+use Throwable;
+
+/**
+ * Class adds Pinterest Conversions API support.
+ */
+class Conversions implements Tracker {
+
+	const EVENT_MAP = array(
+		Tracking::EVENT_PAGE_VISIT    => 'page_visit',
+		Tracking::EVENT_SEARCH        => 'search',
+		Tracking::EVENT_VIEW_CATEGORY => 'view_category',
+		Tracking::EVENT_ADD_TO_CART   => 'add_to_cart',
+		Tracking::EVENT_CHECKOUT      => 'checkout',
+	);
+
+	/**
+	 * @var User User data object. Stores data for Conversions API needs.
+	 */
+	private $user;
+
+	/**
+	 * Pinterest Conversions API class constructor.
+	 *
+	 * @param User $user - User data object to hold ip address and user agent string.
+	 */
+	public function __construct( User $user ) {
+		$this->user = $user;
+	}
+
+	/**
+	 * Track event function implementation. Used to send event data to a destination.
+	 *
+	 * @param string $event_name Tracking event name.
+	 * @param Data   $data       Tracking event data class.
+	 *
+	 * @return void
+	 */
+	public function track_event( string $event_name, Data $data ) {
```

Comment:
Please refactor this method it is too long.

Step 1. Get the $data depending on $event_nam. Something like `get_event_data`.
Step 2. Try block.